walletnode/api/services/multipart.go

```diff
@@ -30,57 +31,68 @@ func UploadImageDecoderFunc(ctx context.Context, service *Artwork) server.Artwor
 	return func(reader *multipart.Reader, p **artworks.UploadImagePayload) error {
 		var res artworks.UploadImagePayload
 
-		for {
-			part, err := reader.NextPart()
-			if err != nil {
-				if err == io.EOF {
-					break
-				}
-				return artworks.MakeInternalServerError(errors.Errorf("could not read next part: %w", err))
-			}
+		filename, err := handleUploadImage(ctx, reader, service.register.Storage)
+		if err != nil {
+			return artworks.MakeBadRequest(err)
```

Comment:
prefer old implementation with more details of type of error - new your changes always return BadRequest error - while some other problems (like storage full is internal server error).